src/ray/gcs/tables.cc

```diff
@@ -172,6 +172,36 @@ Status Log<ID, Data>::CancelNotifications(const JobID &job_id, const ID &id,
                                        pubsub_channel_, nullptr);
 }
 
+template <typename ID, typename Data>
+void Log<ID, Data>::Delete(const JobID &job_id, const ID &id) {
+  RAY_IGNORE_EXPR(GetRedisContext(id)->RunAsync("RAY.TABLE_DELETE", id, /*data=*/nullptr,
+                                                /*length=*/0, prefix_, pubsub_channel_,
+                                                /*redisCallback=*/nullptr));
+}
+
+template <typename ID, typename Data>
+void Log<ID, Data>::Delete(const JobID &job_id, const std::vector<ID> &ids) {
+  if (ids.empty()) {
+    return;
+  }
+  std::unordered_map<RedisContext *, std::ostringstream> sharded_data;
+  for (const auto &id : ids) {
+    sharded_data[GetRedisContext(id).get()] << id.binary();
+  }
+  const size_t batch_size = 1000 * kUniqueIDSize;
+  for (const auto &pair : sharded_data) {
+    std::string current_data = pair.second.str();
+    // Make sure the sending buffer is not too big.
```

Comment:
Move this comment right before the line `const size_t batch_size = 1000 * kUniqueIDSize;` and explain that we break really large deletion commands into batches of size 1000.